google/cloud/aiplatform/prediction/local_model.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from contextlib import contextmanager
+from typing import Optional
+
+from google.cloud import aiplatform
+from google.cloud.aiplatform.prediction import LocalEndpoint
+
+
+class LocalModel:
+    """Class that represents a local model."""
+
+    def __init__(
+        self, serving_container_spec: aiplatform.gapic.ModelContainerSpec,
+    ):
+        """Creates a local model instance.
+
+        Args:
+            serving_container_spec (aiplatform.gapic.ModelContainerSpec):
+                Required. The container spec of the LocalModel instance.
+        """
+        self.serving_container_spec = serving_container_spec
+
+    @contextmanager
+    def deploy_to_local_endpoint(
+        self,
+        artifact_uri: Optional[str] = None,
+        credential_path: Optional[str] = None,
+        host_port: Optional[str] = None,
+        container_ready_timeout: Optional[int] = None,
+        container_ready_check_interval: Optional[int] = None,
+    ):
+        """Deploys the local model instance to a local endpoint.
+
+        Args:
+            artifact_uri (str):
+                Optional. The path to the directory containing the Model artifact and
+                any of its supporting files. Not present for AutoML Models.
+            credential_path (str):
+                Optional. The path to the credential key that will be mounted to the container.
+                If it's unset, the environment variable, GOOGLE_APPLICATION_CREDENTIALS, will
+                be used if set.
+            host_port (str):
+                Optional. The port on the host that the port, AIP_HTTP_PORT, inside the container
+                will be exposed as. If it's unset, a random host port will be assigned.
+            container_ready_timeout (int):
+                Optional. The timeout in second used for starting the container or succeeding the
+                first health check.
+            container_ready_check_interval (int):
+                Optional. The time interval in second to check if the container is ready or the
+                first health check succeeds.
+
+        Returns:
+            A context manager of the local endpoint.
```

Comment:
Since this returns a context manager, let's add an example to the docstring of how it should be used.

Reply:
Done.